src/renderer/vsync/vsync_win.rs

```diff
@@ -1,64 +1,115 @@
 use std::{
     sync::{
         atomic::{AtomicBool, Ordering},
-        Arc, Condvar, Mutex,
+        Arc,
     },
     thread::{spawn, JoinHandle},
+    time::Duration,
 };
-use winapi::um::dwmapi::DwmFlush;
+use winapi::{
+    shared::ntdef::LARGE_INTEGER,
+    um::dwmapi::{DwmGetCompositionTimingInfo, DWM_TIMING_INFO},
+    um::profileapi::{QueryPerformanceCounter, QueryPerformanceFrequency},
+};
+
+use winit::event_loop::EventLoopProxy;
+
+use spin_sleep::SpinSleeper;
 
-use crate::profiling::tracy_zone;
+use crate::{
+    profiling::{tracy_plot, tracy_zone},
+    window::UserEvent,
+};
 
 pub struct VSyncWin {
     should_exit: Arc<AtomicBool>,
     vsync_thread: Option<JoinHandle<()>>,
```

Comment:
I guess you left that since it's not really affected by this PR, but `vsync_thread` seems to be never `None`, so the `Option` wrapper could be removed.

Reply:
I'm not sure if there's a better way, but the problem is that the join handle is not valid after joining, so there's problems with the ownership.

Using an option seem to be a quite common recommendation, here https://stackoverflow.com/questions/69189818/why-can-t-i-use-a-thread-handle-which-is-stored-in-a-struct and here https://users.rust-lang.org/t/joinhandle-join/81874/3

Reply:
Sounds good!